rules-reviewed/eap8/eap7/javaee-to-jakarta-namespaces.windup.xml

```diff
@@ -567,5 +568,512 @@
+                        </quickfix>
+                    </hint>
+                </iteration>
+            </perform>
+            <where param="quotes">
+                <matches pattern="(&quot;|')" />
+            </where>
+        </rule>
+        <rule id="javaee-to-jakarta-namespaces-00042">
+            <when>
+                <xmlfile matches="p:ejb-jar[@version='3.2']" as="xml-file">
+                    <namespace prefix="p" uri="http://xmlns.jcp.org/xml/ns/javaee" />
+                </xmlfile>
+                <filecontent pattern="version={quotes}3.2{quotes}" from="xml-file" as="result"/>
+            </when>
+            <perform>
+                <iteration over="result">
+                    <hint title="Replace the Java EE version with the Jakarta equivalent" effort="1" category-id="mandatory">
+                        <message>`ejb-jar_3_2.xsd`: In the root tag, replace the `version` attribute value with `4.0`</message>
+                        <link title="Jakarta XML Schemas" href="https://jakarta.ee/xml/ns/jakartaee/#9" />
+                        <quickfix type="REPLACE" name="javaee-namespaces-replace-ejb-jar_3_2-version">
+                            <replacement>4.0</replacement>
+                            <search>3.2</search>
+                        </quickfix>
+                    </hint>
+                </iteration>
+            </perform>
+            <where param="quotes">
+                <matches pattern="(&quot;|')" />
+            </where>
+        </rule>
+        <rule id="javaee-to-jakarta-namespaces-00043">
+            <when>
+                <xmlfile matches="p:webservices[@version='1.4']" as="xml-file">
+                    <namespace prefix="p" uri="http://xmlns.jcp.org/xml/ns/javaee" />
+                </xmlfile>
+                <filecontent pattern="version={quotes}1.4{quotes}" from="xml-file" as="result"/>
+            </when>
+            <perform>
+                <iteration over="result">
+                    <hint title="Replace the Java EE version with the Jakarta equivalent" effort="1" category-id="mandatory">
+                        <message>`javaee_web_services_1_4.xsd`: In the root tag, replace the `version` attribute value with `2.0`</message>
+                        <link title="Jakarta XML Schemas" href="https://jakarta.ee/xml/ns/jakartaee/#9" />
+                        <quickfix type="REPLACE" name="javaee-namespaces-replace-javaee_web_services_1_4-version">
+                            <replacement>2.0</replacement>
+                            <search>1.4</search>
+                        </quickfix>
+                    </hint>
+                </iteration>
+            </perform>
+            <where param="quotes">
+                <matches pattern="(&quot;|')" />
+            </where>
+        </rule>
+        <rule id="javaee-to-jakarta-namespaces-00044">
+            <when>
+                <xmlfile matches="p:web-app[@version='3.1']" as="xml-file">
```

Comment:
Hey Carlos, I'm quite a newbie so maybe I'm wrong here, but shouldn't this be (according to [this spreadsheet](https://docs.google.com/spreadsheets/d/1xYG_0f-vrD1D90sDXFH63krmKTXKbNClADTcXsXQWu0/edit#gid=0)) version `4.0` instead of `3.1`?

```suggestion
                <xmlfile matches="p:web-app[@version='4.0']" as="xml-file">
```

Reply:
Fair enough, just saw that the suggested change is in another rule downthere :+1:

Reply:
Yes, as you have pointed out the `version='4.0'` you are looking for, is just below the Rule this "Suggested change" is pointing https://github.com/windup/windup-rulesets/blob/8c9fd410d2c59cefd7521170729ec10b372e3443/rules-reviewed/eap8/eap7/javaee-to-jakarta-namespaces.windup.xml#L802